js/objects/System.js

```diff
@@ -562,27 +556,11 @@ const System = {
             this.tagNameForViewNamed(partName)
         );
         newView.setModel(model);
-        // if the new model is a Card we make sure add it after the last card
-        // on the stack
-        if((newView.name === 'CardView' || newView.name === 'StackView') && parentElement.childNodes.length){
-            let lastNode;
-            parentElement.childNodes.forEach((child) => {
-                if(child.name === newView.name){
-                    lastNode = child;
-                }
-            });
-            lastNode.after(newView);
-        } else {
-            if(parentElement.type == "window"){
-                // slot the new view into the window
-                let parentViews = this.findViewsById(parentElement.id);
-                parentViews.forEach((window) => {
-                    let pane = window._shadowRoot.querySelector('.st-window-pane');
-                    pane.append(newView);
-                });
-            }
-            parentElement.appendChild(newView);
-        }
+        this.sendMessage({
+            type: "viewChanged",
+            changeName: "subpart-new",
+            args: [newView]
```

Comment:
One thing we said early on is that we wanted messages to be serializable. Is that still the case? If so, we need to make the argument passed with this message to be a part id. The element lookup can be handles by any corresponding view that receives the message.

Reply:
@darth-cheney you mean?

```
this.sendMessage({
type: "viewChanged",
changeName: "subpart-new",
partId: ID
args: [newView]},
target)?
```

Reply:
`newView` is a DOM element here, which is not technically serializable (unless we want to serialize it's HTML). That is the argument that should be an id instead. Then the receiver can lookup the corresponding element based on the id as needed.